src/components/guestsList/guestsList.js

```diff
@@ -0,0 +1,243 @@
+ *
+ * @augments React.Component
+ * @fires onUpdateGuestsData
+ * @fires onScroll
+ */
+class GuestsList extends React.Component {
+  state = { currentPage: 0, limit: 10, previousData: [] };
+
+  componentDidMount() {
+    this.onUpdateGuestsData();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { currentPage } = this.state;
+
+    if (currentPage !== prevState.currentPage) {
+      this.onUpdateGuestsData();
+    }
+  }
+
+  /**
+   * Call the RHSM APIs, apply filters.
+   *
+   * @event onUpdateGuestsData
+   */
+  onUpdateGuestsData = async () => {
+    const { currentPage, limit } = this.state;
+    const { getHostsInventoryGuests, query, id } = this.props;
+
+    if (id) {
+      const updatedQuery = {
+        ...query,
+        [RHSM_API_QUERY_TYPES.LIMIT]: limit,
+        [RHSM_API_QUERY_TYPES.OFFSET]: currentPage * limit || 0
+      };
+
+      const { guestsQuery } = apiQueries.parseRhsmQuery(updatedQuery);
+      await getHostsInventoryGuests(id, guestsQuery);
+    }
+  };
+
+  /**
+   * Update page state.
+   *
+   * @event onScroll
+   * @param {object} event
+   */
+  onScroll = event => {
+    const { target } = event;
+    const { currentPage, limit, previousData } = this.state;
+    const { numberOfEntries, pending, listData } = this.props;
+
+    const bottom = target.scrollHeight - target.scrollTop === target.clientHeight;
+
+    if (numberOfEntries > (currentPage + 1) * limit && bottom && !pending) {
+      const newPage = currentPage + 1;
+      const updatedData = [...previousData, ...(listData || [])];
+
+      this.setState({
+        previousData: updatedData,
+        currentPage: newPage
+      });
+    }
+  };
+
+  renderLoader() {
+    const { currentPage } = this.state;
+    const { filterGuestsData, listData, pending } = this.props;
+
+    if (currentPage > 0 && pending) {
+      const scrollLoader = (
+        <Loader
+          variant="table"
+          tableProps={{
+            borders: false,
+            colCount: filterGuestsData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
+            rowCount: 0,
+            variant: TableVariant.compact
+          }}
+        />
+      );
+
+      return <div className="curiosity-table-scroll-loader__custom">{scrollLoader}</div>;
+    }
+
+    return null;
+  }
+
+  /**
+   * Render a guests table.
+   *
+   * @returns {Node}
+   */
+  renderTable() {
+    const { previousData } = this.state;
+    const { filterGuestsData, listData } = this.props;
+    let updatedColumnHeaders = [];
+
+    const updatedRows = [...previousData, ...(listData || [])].map(({ ...cellData }) => {
+      const { columnHeaders, cells } = inventoryListHelpers.parseRowCellsListData({
+        filters: filterGuestsData,
+        cellData
+      });
+
+      updatedColumnHeaders = columnHeaders;
+
+      return {
+        cells
+      };
+    });
+
+    return (
+      <div className="curiosity-table-scroll" style={{ height: `200px` }}>
```

Comment:
Required:
This height should be dynamic based on a max and min number of rows. If possible use CSS calc to help generate this number. If not possible it appears an average of "42" pixels is currently leveraged by each row